File: 2884/2884.java
import java.util.Scanner;

class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        int hour = sc.nextInt();
        int minute = sc.nextInt();
        
        int before45 = minute - 45;
        
        if (before45 < 0) {
            hour = (hour - 1 < 0) ? 23 : hour - 1 ; // 삼항 연산자로 1 뺐을 때 음수이면 23 , 아니면 hour -1 
            minute = 60 + before45;
        } else {
            minute -= 45;
        }
        
        System.out.println(hour+ " "+ minute);
        
    }
}

File: 2884/2884.rb
time = gets().split(' ')

hour = time[0].to_i
minute = time[1].to_i

before45 = minute - 45

if before45 < 0
    hour -= 1
    minute = 60 + before45
    hour = 23 if hour < 0  # 코드가 복잡해지는게 싫어서 조건문을 축약해서 사용 
else
    minute -= 45
end

puts("#{hour} #{minute}") # puts(hour.to_s + " " + minute.to_s) 로 쓸 수도 있지만 syntax 로 사용하는게 더 간단